naga/src/front/wgsl/parse/mod.rs

```diff
@@ -296,6 +301,19 @@ impl Parser {
         )
     }
 
+    fn track_recursion<'a, F, R>(&mut self, f: F) -> Result<R, Error<'a>>
+    where
+        F: FnOnce(&mut Self) -> Result<R, Error<'a>>,
+    {
+        self.recursion_depth += 1;
+        if self.recursion_depth >= 256 {
```

Comment:
thought: We already have some recursion tracking for curly braces. I wonder how they might be reconciled.

Reply:
That is definitely something to consider, but I would like to treat it as a follow-up. We have particular reasons that #5757 is urgent to close despite being not very important to users, so we want to put this behind us quickly.

Reply:
I agree; generally, [`thought`s I offer are not blocking](https://conventionalcomments.org/), so 👍🏻 if it's a problem, weeee'll figure it out!